src/compat/should_wait_for_data_before_loaded.ts

```diff
@@ -25,7 +25,14 @@ import { isSafariMobile } from "./browser_detection";
  * @returns {Boolean}
  */
 export default function shouldWaitForDataBeforeLoaded(
-  isDirectfile : boolean
+  isDirectfile : boolean,
+  mustPlaysInline: boolean
 ) : boolean {
-  return !isDirectfile || !isSafariMobile;
+  if (isDirectfile) {
+    if (isSafariMobile && mustPlaysInline) {
+      return true;
+    }
+    return false;
+  }
+  return !(isSafariMobile && mustPlaysInline);
```

Comment:
Maybe it's simpler :
```
if (isDirectFile) {
    return isSafariMobile && mustPlaysInLine;
}
return !(isSafariMobile && mustPlaysInLine);
```

Or, even if it is much less readable, we could write :
`return (isDirectFile === isSafariMobile && mustPlaysInLine);`

Reply:
I found a way to be clearer, thanks :)